src/Detector/Node/NodeDetector.cs

```diff
@@ -211,15 +213,20 @@ private IEnumerable<FrameworkInfo> DetectFrameworkInfos(DetectorContext context)
 
                 // wild-card dependency
                 (bool isWildCardDependency, string wildCarddependencyName) = GetWildCardDependency(dependencyName);
-
-                if (monitoredDevDependencies.ContainsKey(dependencyName) || isWildCardDependency) 
+                
+                bool isValidDependency = monitoredDevDependencies.ContainsKey(dependencyName);
+                string frameworkName = isValidDependency ? monitoredDevDependencies[dependencyName] : "";
+                frameworkName = isWildCardDependency ? wildCarddependencyName : frameworkName;
+                bool isNotDuplicateFramework= !(frameworksSet.Contains(frameworkName));
```

Comment:
nit: to avoid confusion when using it in a conditional, I typically steer away from using `Not` (or other negation words) in `bool` variable names.

In this case, I would either keep the right-hand side logic the same and rename the variable to `isNewFramework`, or drop the leading `!`, rename the variable to `isDuplicateFramework` and change the conditional below to `!isDuplicateFramework`.

Reply:
This piece got removed